### ISSUE TITLE
Neo-cli Unable to Sync Blocks

### ISSUE THREAD

**Issue by thehomosapien** [neo-project/neo-cli#434]
OS: Ubuntu 16.04.06

NEO-CLI Version: 2.10.2.0

1. Loaded plugins:
------------------
ApplicationLogs
ImportBlocks
RpcWallet
RpcSecurity
SimplePolicyPlugin
CoreMetrics
RpcNep5Tracker
RpcSystemAssetTrackerPlugin

2. config.json:
---------------
                                                      

{
  "ApplicationConfiguration": {
    "Paths": {
      "Chain": "Chain_{0}",
      "Index": "Index_{0}"
    },
    "P2P": {
      "Port": 10333,
      "WsPort": 10334
    },
    "RPC": {
      "BindAddress": "0.0.0.0",
      "Port": 3007,
      "SslCert": "",
      "SslCertPassword": ""
    },
    "UnlockWallet": {
      "Path": "/home/ubuntu/.neo/Xbond_Neo_wallet.db3",
      "Password": "somepassword",
      "StartConsensus": false,
      "IsActive": true
    },
    "PluginURL": "https://github.com/neo-project/neo-plugins/releases/download/v{1}/{0}.zip"
  }
}

I downloaded the source code and complied it, I got all the binaries using Readme.md at https://github.com/neo-project/neo-cli

Issue: 
-------
1. My chain isn't getting over index 328296

2. Whenever I run ./neo-cli, after few seconds it flashes this:

 [ERROR][8/13/19 7:17:03 AM][Thread 0003][akka://NeoSystem/user/$a] Object reference not set to an instance of an object.
Cause: System.NullReferenceException: Object reference not set to an instance of an object.
   at Neo.Plugins.LogReader.OnPersist(Snapshot snapshot, IReadOnlyList`1 applicationExecutedList)
   at Neo.Ledger.Blockchain.Persist(Block block)
   at Neo.Ledger.Blockchain.OnNewBlock(Block block)
   at Neo.Ledger.Blockchain.OnReceive(Object message)
   at Akka.Actor.UntypedActor.Receive(Object message)
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
   at Akka.Actor.ActorCell.ReceiveMessage(Object message)
   at Akka.Actor.ActorCell.Invoke(Envelope envelope)
[ERROR][8/13/19 7:17:03 AM][Thread 0008][akka://NeoSystem/user/$a] Error while creating actor instance of type Neo.Ledger.Blockchain with 2 args: (Neo.NeoSystem,Neo.Persistence.LevelDB.LevelDBStore)
Cause: [akka://NeoSystem/user/$a#1863269873]: Akka.Actor.PostRestartException: Exception post restart (System.NullReferenceException) ---> System.TypeLoadException: Error while creating actor instance of type Neo.Ledger.Blockchain with 2 args: (Neo.NeoSystem,Neo.Persistence.LevelDB.LevelDBStore) ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at Neo.Ledger.Blockchain..ctor(NeoSystem system, Store store)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at Akka.Actor.Props.ActivatorProducer.Produce()
   at Akka.Actor.Props.NewActor()
   --- End of inner exception stack trace ---
   at Akka.Actor.Props.NewActor()
   at Akka.Actor.ActorCell.CreateNewActorInstance()
   at Akka.Actor.ActorCell.<>c__DisplayClass109_0.<NewActor>b__0()
   at Akka.Actor.ActorCell.UseThreadContext(Action action)
   at Akka.Actor.ActorCell.NewActor()
   at Akka.Actor.ActorCell.FinishRecreate(Exception cause, ActorBase failedActor)
   --- End of inner exception stack trace ---

**Comment by shargon** [comment]
If you remove the plugins: `ApplicationLogs`, `RpcNep5Tracker` and `RpcSystemAssetTrackerPlugin`, do you have the same issue?
Did you tried without plugins?

**Comment by thehomosapien** [comment]
Hey thanks for pointing out the plugins. I solved the issue in a different way and it seems it's bug. When we import a plugin from neo-cli using install <pluginName>, it creates a directory by the name of that specific plugin inside the Plugin directory. 

"neo-cli install ImportBlocks" creates this:
Plugins/ImportBlocks/Plugins/ImportBlocks/config.json and
Plugins/ImportBlocks/Plugins/ImportBlocks/ImportBlocks.dll

While according to https://github.com/neo-project/neo-plugins Readme.md the correct way is this: 
./Plugins/ImportBlocks.dll
./Plugins/ImportBlocks/config.json

I had to moved the dll files earlier but this time I moved the config.json files for each Plugin to the right path. Now it seems to work fine. But surely it's a bug with install <plugin> command.

Again thanks @shargon  for all the help.
